### ISSUE TITLE
CI infrastructure: error during connect: docker.sock: EOF

### ISSUE THREAD

**Issue by tavplubix** [ClickHouse/ClickHouse#43084]
```
error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=clickhouse%2Fbinary-builder&tag=latest": EOF
2022-11-08 15:15:29,455 Cannot pull image clickhouse/binary-builder:latest
2022-11-08 15:15:29,455 Will build image with cmd: 'docker build --network=host -t clickhouse/binary-builder:latest -f /home/ubuntu/actions-runner/_work/_temp/build_check/ClickHouse/docker/packager/../../docker/packager/binary/Dockerfile /home/ubuntu/actions-runner/_work/_temp/build_check/ClickHouse/docker/packager/../../docker/packager/binary'
error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/build?buildargs=%7B%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=host&rm=1&shmsize=0&t=clickhouse%2Fbinary-builder%3Alatest&target=&ulimits=null&version=1": EOF
Traceback (most recent call last):
  File "/home/ubuntu/actions-runner/_work/_temp/build_check/ClickHouse/docker/packager/./packager", line 419, in <module>
    build_image(image_with_version, dockerfile)
  File "/home/ubuntu/actions-runner/_work/_temp/build_check/ClickHouse/docker/packager/./packager", line 37, in build_image
    subprocess.check_call(
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'docker build --network=host -t clickhouse/binary-builder:latest -f /home/ubuntu/actions-runner/_work/_temp/build_check/ClickHouse/docker/packager/../../docker/packager/binary/Dockerfile /home/ubuntu/actions-runner/_work/_temp/build_check/ClickHouse/docker/packager/../../docker/packager/binary' returned non-zero exit status 1.
```
https://s3.amazonaws.com/clickhouse-builds/42878/eb19df0094bc6bda65d5b39c222fb8c501d02f75/clickhouse_special_build_check/report.html

**Comment by tavplubix** [comment]
https://s3.amazonaws.com/clickhouse-builds/39737/dbf64e27125bcd4e4882d22957695d466dde7152/clickhouse_special_build_check/report.html

**Comment by Felixoid** [comment]
It looks like I haven't updated the runners after https://github.com/ClickHouse/ClickHouse/pull/42082 was merged, so we don't restart docker on issues

**Comment by Felixoid** [comment]
Updated it 10 minutes ago, it should at least kick-off the dead host pretty quickly, or restart the docker at least

**Comment by tavplubix** [comment]
Similar issue in backport PR: https://s3.amazonaws.com/clickhouse-test-reports/43807/812e08250d3aaaca56875d7d66f12697cb100c28/integration_tests__release_/main_script_log.txt

**Comment by Felixoid** [comment]
Interestingly, the docker couldn't start on the brand-new instance, and then it was terminated at the end of the running

![image](https://user-images.githubusercontent.com/3025537/204813952-908a50cf-66ee-464d-80bb-1117a3b22e25.png)

So, at least the part about killing bad hosts works. Not sure yet what could we do regarding so floating issue as `docker doesn't start on the boot` though

**Comment by qoega** [comment]
Is it useful to run trivial checks like docker hello world, checkout public repo, download s3 file before starting GitHub worker? We can just reboot/shutdown host on failure and no CI checks will see errors.

**Comment by Felixoid** [comment]
It may actually help. Just executing https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/worker/init_runner.sh#L60 before the starting the runner

**Comment by tavplubix** [comment]
It happened again: https://s3.amazonaws.com/clickhouse-test-reports/48417/f7586e38ed3c063fa11d6783e5dbb2a47ca53f8d/integration_tests__asan__[1/6].html

**Comment by Felixoid** [comment]
It's 99.9% a dying runner, the next snippets are from the [log](https://s3.amazonaws.com/clickhouse-test-reports/48417/f7586e38ed3c063fa11d6783e5dbb2a47ca53f8d/integration_tests__asan__[1/6]/main_script_log.txt)

```
# the docker works first
2023-04-05 13:00:42,000 [ 2003258 ] INFO : src dir is not set. Will use /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/ClickHouse/src (runner:96, check_args_and_update_paths)
2023-04-05 13:00:42,000 [ 2003258 ] INFO : base_configs_dir: /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/ClickHouse/programs/server, binary: /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/build/clickhouse, cases_dir: /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/ClickHouse/tests/integration  (runner:98, check_args_and_update_paths)
2023-04-05 13:00:42,000 [ 2003258 ] INFO : Unknown image clickhouse/integration-helper (runner:346, <module>)
Pulling redis1 ... 
Pulling redis1 ... pulling from library/redis
Pulling redis1 ... digest: sha256:7b83a0167532d4320a...
Pulling redis1 ... status: image is up to date for r...
Pulling redis1 ... done
# but then, in three minutes
2023-04-05 13:03:00,627 [ 2018665 ] INFO : base_configs_dir: /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/ClickHouse/programs/server, binary: /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/build/clickhouse, cases_dir: /home/ubuntu/actions-runner/_work/_temp/integration_tests_asan/ClickHouse/tests/integration  (runner:98, check_args_and_update_paths)
2023-04-05 13:03:00,627 [ 2018665 ] INFO : Unknown image clickhouse/integration-helper (runner:346, <module>)
error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/volumes/create": EOF
error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/volumes": EOF
error during connect: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json?all=1&filters=%7B%22name%22%3A%7B%22clickhouse_integration_tests_xsi18t%22%3Atrue%7D%7D": EOF
```

And the [job](https://github.com/ClickHouse/ClickHouse/actions/runs/4616770150/jobs/8165309040) is failed to finish.

Did you merge without the restart? It looks to me like it wouldn't be restarted because we already have the report.

So far the only костыль I could think about is parsing `main_script_log.txt` for the obvious infrastructure errors like `all 'error during connect', 'docker.sock', 'EOF' in line for lines in log`.

Thoughts?

**Comment by Felixoid** [comment]
Should be addressed by #49283, waiting some time if it will be reproduced

**Comment by Felixoid** [comment]
I am closing it for now, although feel free to reopen on the same issues
